### ISSUE TITLE
Function switch

### ISSUE THREAD

**Issue by TMO2018** [tebl/c64-blinkendiag#2]
The BlinkenDiag (rev B) does not work when the FUNCTION switch is pressed and remains in RESET. GAME is activated. I swapped the Eeprom and the switch and checked the wiring and couldn't find any faults. Do you have any idea?
I've tested on a 407 and 469 board.

**Comment by tebl** [comment]
I'm not sure, it shouldn't stay in reset. Does that mean you've tested it
and the reset is actively held low, as measured with multimeter, or just
that it fails to boot?

The biggest gotcha is usually the switches, they seem to have a different
pinout depending on the colour on the bottom of them (it should be black).
Preferably, test with multimeter to see that they're working as indicated
on the silkscreen.

Also check that it isn't mounted upside down, I seem to remember this made
a difference on a previous build though I might be remembering the wrong
switch.

tor. 15. apr. 2021, 20:35 skrev TMO2018 ***@***.***>:

**Comment by TMO2018** [comment]
I tested the switches with a multimeter and they corresponded to the function of the circuit diagram. The switches have a black base and a light gray top.
I have measured the conductor tracks to the expansion port and Eeprom including the different switch positions and it looks right.
When the function switch is pressed, the C64 remains stuck in reset. The reset LED lights up constantly, just like the game LED (U2: 74ACT540). The same behavior with a 74HCT541, but with inverted LEDs.
The Kernals C64-Diag and 1541-Diag always work. At first, deadtest and stid deadtest did not work. Then only once and then never again. I haven't been able to reproduce this behavior until now. That's what I'm a little confused about right now.
When I try a CHECK64 module, all four tests work (Deadtest, C64-Diag, 1541-Test, C128-Diag).

**Comment by tebl** [comment]
The function switch mainly just switches around two of the signals, so they
shouldn't really affect the reset line. GAME and RESET are both present on
U2, pins 2 and 3, so at this point they are next to each other and there
might be a small bridge there.

If you have another set of PCBs, it helps to have it next to you when
troubleshooting - makes it easier to beep out which things are on the PCB
and which are in contact due to the components.

fre. 16. apr. 2021, 10:43 skrev TMO2018 ***@***.***>:

**Comment by TMO2018** [comment]
I will get a second board in the next few days and will continue to search for the error. Thank you for your assistance.

**Comment by TMO2018** [comment]
I found the error thanks to your reference to pins 2 and 3 of U2. I unsoldered the IC socket from U2. The dead test works without the IC socket. I could hardly believe it, but there was an electrical connection between pins 2 and 3 in the unsoldered IC socket. I couldn't see it with my eyes. I soldered in a new socket and now the cartrige works as expected. Thank you for your support. That was very helpful.
